File: cpp/medium/EvaluateReversePolishNotation.cpp
#include <charconv>
#include <iostream>
#include <stack>
#include <string>
#include <vector>
using namespace std;

class Solution {
  stack<int> st;
  int ans = 0;

public:
  int evalRPN(vector<string> &tokens) {
    for (int i = 0; i < tokens.size(); i++) {
      if (tokens[i] == "+" and not st.empty()) {
        int first = st.top(), second;
        st.pop();
        if (st.empty()) {
          continue;
        }
        second = st.top();
        st.pop();
        this->ans += (first + second);
      } else if (tokens[i] == "-" and not st.empty()) {
        int first = st.top(), second;
        st.pop();
        if (st.empty()) {
          continue;
        }
        second = st.top();
        st.pop();
        this->ans += (first - second);
      } else if (tokens[i] == "/" and not st.empty()) {
        int first = st.top(), second;
        st.pop();
        if (st.empty()) {
          continue;
        }
        second = st.top();
        st.pop();
        this->ans += (first / second);
      } else if (tokens[i] == "*" and not st.empty()) {
        int first = st.top(), second;
        st.pop();
        if (st.empty()) {
          continue;
        }
        second = st.top();
        st.pop();
        this->ans += (first * second);
      } else {
        st.push(stoi(tokens[i]));
      }
    }
    return this->ans;
  }
};

int main() {}

File: ts/medium/EvaluateReversePolishNotation.js

class Solution {
	/**
	 * @param {string[]} tokens
	 * @return {number}
	 */
	evalRPN(tokens) {
		const stack = [];
		for (const token of tokens) {
			if (token === "+") {
				const first = stack.pop();
				const second = stack.pop()
				stack.push(second + first);
			} else if (token === "-") {
				const first = stack.pop();
				const second = stack.pop()
				stack.push(second - first);
			} else if (token === "/") {
				const first = stack.pop();
				const second = stack.pop()
				stack.push(Math.trunc(second / first));
			} else if (token === "*") {
				const first = stack.pop();
				const second = stack.pop()
				stack.push(second * first);
			} else {
				stack.push(parseInt(token));
			}

		}
		return stack.pop();
	}
}
